### PR TITLE
[WIP] Feature/repa

### PR DESCRIPTION
**Pull request by LouisRouss**
Add repa training possibility

Done:
- Repa proj in DiT
- DinoV2 encoder added
- Add VAE abstraction

Left to do:
- Add some VAE classes (SDXL, MAETok etc...)
- Add repa loss to both ddpm and flow matching
- test everything

### CODE REVIEW

File: src/diffulab/training/losses/repa.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        if dst_features is None:
+            assert self.repa_encoder is not None, "REPA encoder must be initialized to compute features."
+            with torch.no_grad():
+                dst_features = self.repa_encoder(
+                    x0
+                )  # batch size seqlen embedding_dim # SEE HOW TO HANDLE THE PRE COMPUTING OF FEATURES
+        assert dst_features is not None, "Destination features must be provided or computed."
+
+        projected_src_features: Tensor = self.proj(self.src_features)
+        if self.resampler is not None:
```
Comment (LouisRouss) [inline src/diffulab/training/losses/repa.py:97]: Hummm I don't see any reason for the latent to not degenerate and the training to not collapse, should probably add a regularization loss
